omp/reorder/rcm_kernels.cpp

```diff
@@ -382,9 +368,10 @@ std::pair<IndexType, IndexType> rls_contender_and_height(
 
 #pragma omp for schedule(static)
         for (IndexType i = 1; i < num_vertices; ++i) {
-            if (std::tie(levels[i], degrees[i]) >
-                std::tie(local_contender.first.first,
-                         local_contender.first.second)) {
+            if (levels[i] != std::numeric_limits<IndexType>::max() &&
+                std::tie(levels[i], degrees[i]) >
```

Comment:
this seems to be choosing a max degree node instead of a min degree node? cc @lksriemer

Reply:
I fixed the issue, but it's pretty hard to test, so I'll just leave it as-is.